ts/private/ts_project_worker.js

```diff
@@ -353,7 +353,8 @@ function createProgram(args, initialInputs) {
     const filesystemTree = createFilesystemTree(execRoot, initialInputs);
     const outputs = new Set();
 
-    const taskQueue = new Array();
+    // NOTE: tsc will never clearTimeout if the index is 0 hence array needs to have at least one item.
```

Comment:
Why? Who does that?

Reply:
Right. I should put a link there. https://github.com/microsoft/TypeScript/blob/d0bfd8caed521bfd24fc44960d9936a891744bb7/src/compiler/watchPublic.ts#L681 
if we return zero from `setTimeout` then this condition will never be met thus leading to incorrect behavior.

Reply:
But this is the array index, not the timeout id..?

Reply:
Ohh nm, I see it now that I look at the full file 👍

Reply:
Can we just change `setTimeout` to not have the `-1`?

Reply:
it won't have `-1` it'll have `0` which is the problem here.

Reply:
> But this is the array index, not the timeout id..?

for the `setTimeout`, it is the index but for tsc, it is a number.

Reply:
I'd think just returning `+1` and deleting `-1` within the set/clear would be easier, then all the "id" logic is within those methods and this array doesn't care. Those methods could change to persist `{id: Math.random(), callback}` and this array wouldn't care etc...